src/mecca/lib/memory.d

```diff
@@ -441,8 +442,19 @@ struct DRuntimeStackDescriptor {
     DRuntimeStackDescriptor*    prev;
 
     static assert (__traits(classInstanceSize, Mutex) == 72); // This size is part of the mangle
-    pragma(mangle, "_D4core6thread6Thread6_locksG2G72v") extern __gshared static
+    static if (__traits(hasMember, Thread, "_locks")) {
+        pragma(mangle, "_D4core6thread6Thread6_locksG2G72v") extern __gshared static
             void[__traits(classInstanceSize, Mutex)][2] _locks;
+        @notrace private Mutex _slock() nothrow @nogc {
+            return cast(Mutex)_locks[0].ptr;
+        }
+    } else {
+        pragma(mangle,"_D4core6thread6Thread6_slockG72v") extern __gshared static
+            void[__traits(classInstanceSize, Mutex)] _slock;
+        @notrace private Mutex _slock() nothrow @nogc {
```

Comment:
@thewilsonator How does this work? Aren't there two members with the same name (`_slock`) within the same scope?